api/utils/aws/identifiers.go

```diff
@@ -180,6 +202,16 @@ var (
 	// > special characters other than underscore (_) are not supported
 	matchGlueName = regexp.MustCompile(`^[a-z0-9_]{1,255}$`).MatchString
 
+	// matchRolesAnywhereTrustAnchorName is a regex that matches against AWS IAM Roles Anywhere Trust Anchor Names.
+	// See https://docs.aws.amazon.com/rolesanywhere/latest/APIReference/API_CreateTrustAnchor.html#API_CreateTrustAnchor_RequestBody
+	matchRolesAnywhereTrustAnchorName = baseResourceNameMatcher
+
+	// matchRolesAnywhereProfileName is a regex that matches against AWS IAM Roles Anywhere Trust Anchor Names.
+	// See https://docs.aws.amazon.com/rolesanywhere/latest/APIReference/API_CreateProfile.html#API_CreateProfile_RequestBody
+	matchRolesAnywhereProfileName = baseResourceNameMatcher
+
+	baseResourceNameMatcher = regexp.MustCompile(`^[ a-zA-Z0-9-_]{1,255}$`).MatchString
```

Comment:
Could we make it stricter still? In particular allowing spaces leads to degenerate cases like `     ` or `  foo bar    ` makes me uneasy.

Reply:
This is not a unique identifier.
AWS accepts names with spaces for those resources:
<img width="242" alt="image" src="https://github.com/user-attachments/assets/32a300ce-83a4-448c-a264-7dc85037bb10" />

I agree with you that adding spaces to the mix usually leads to subtle bugs or UI issues.
Here's an example with the profile above in AWS UI:
Page title kept the space at the end
![image](https://github.com/user-attachments/assets/76a00d6f-2ddb-41b7-8ce2-46c2b9452bc3)

Resource detail remove both spaces:
![image](https://github.com/user-attachments/assets/65ef5a70-3eb2-4033-b2fd-d171dfd12f33)


I don't think our validation should be stricter than AWS.